projects/plugins/jetpack/modules/subscriptions/jetpack-user-content-link-redirection.php

```diff
@@ -48,6 +69,29 @@ function jetpack_user_content_link_redirection() {
 	exit;
 }
 
+/**
+ * Check if the domain of the request is the same as the domain of the given blog_id
+ *
+ * @param string $domain_from_request The domain from the request.
+ * @param int    $blog_id             The blog_id to check against.
+ *
+ * @return bool
+ */
+function jetpack_user_content_link_is_same_domain( $domain_from_request, $blog_id ) {
+	$request  = sprintf( '/sites/%d?fields=URL', $blog_id );
+	$response = Client::wpcom_json_api_request_as_blog( $request );
+	if ( is_wp_error( $response ) ) {
```

Comment:
`wp_remote_retrieve_body()` ([docs](https://developer.wordpress.org/reference/functions/wp_remote_retrieve_body/)) also checks for `is_wp_error( $response )` and returns an empty string if both the response or the response body is not in the correct shape. We can leverage that and simplify the checking here.

The use of `json_decode( wp_remote_retrieve_body( $response ) )->URL` feels very unsafe as `wp_remote_retrieve_body( $response )` could return an empty string, and also this line does not cater for the possiblity of there the `URL` property doesn't exist because `json_decode` was decoding an empty string, or simply because the response did not include the `URL` property.